contrib/icinga2-commands.conf

```diff
@@ -8,81 +8,198 @@ template CheckCommand "cloud_aws" {
         }
         "-a" = {
-            value = "$cloud_aws_autoscale$"
+            value = "$cloud_aws_ec2_instances_autoscale$"
             description = "Search for ec2 instances by autoscaling group"
         }
     }
+}
+
+object CheckCommand "cloud_aws_cloudfront" {
+    import "cloud_aws"
+
+    command += [ "cloudfront" ]
+
+    arguments += {
+        "-e" = {
+            value = "$cloud_aws_etag$"
+            description = "Etag name of one or multiple distributions. If --etag is empty, all distributions will be evaluated"
+        }
+    }
+}
+
+object CheckCommand "cloud_aws_s3_bucket" {
+    import "cloud_aws"
+
+    command += [ "s3", "bucket" ]
 
-    vars.cloud_aws_ec2 = "ec2"
-    vars.loud_aws_instances = "instances"
+    arguments += {
+        "-b" = {
+            value = "$cloud_aws_s3_bucket_buckets$"
+            repeat_key = true
+            description = "Name of the S3 bucket. If --buckets is empty, all buckets will be evaluated"
+        }
+        "-c" = {
+            value = "$cloud_aws_s3_bucket_crit_bucket_size$"
+            description = "Critical threshold for the size of the specified bucket. Alerts if the size is greater than the critical threshold"
+        }
+        "-w" = {
+             value = "$cloud_aws_s3_bucket_warn_bucket_size$"
+             description = "Warning threshold for the size of the specified bucket. Alerts if the size is greater than the warning threshold"
+        }
+        "--empty-ok" = {
+            set_if = "$cloud_aws_s3_bucket_empty_ok$"
+            description = "Return OK if there are no buckets"
+        }
+    }
 }
 
-apply Service "cloud_aws_single_instance" {
+object CheckCommand "cloud_aws_s3_object" {
+    import "cloud_aws"
+
+    command += [ "s3", "object" ]
+
+    arguments += {
+        "-b" = {
+            value = "$cloud_aws_s3_object_bucket$"
+            repeat_key = true
+            description = "Name of the S3 bucket. If --buckets is empty, all buckets will be evaluated"
+        }
+        "-c" = {
+            value = "$cloud_aws_s3_object_crit_bucket_size$"
+            description = "Critical threshold for the size of the specified bucket. Alerts if the size is greater than the critical threshold"
+        }
+        "-w" = {
+             value = "$cloud_aws_s3_object_warn_bucket_size$"
+             description = "Warning threshold for the size of the specified bucket. Alerts if the size is greater than the warning threshold"
+        }
+        "--prefix" = {
+            value = "$cloud_aws_s3_object_prefix$"
+            description = "Limits the response to keys that begin with the specified prefix"
+        }
+        "-p" = {
+            set_if = "$cloud_aws_s3_object_perfdata$"
+            description = "Displays perfdata and lists all objects in the specified bucket"
+        }
+        "--empty-ok" = {
+            set_if = "$cloud_aws_s3_object_empty_ok$"
+            description = "Return OK if there are no buckets"
+        }
```

Comment:
Would defaults for perfdata handling and empty make sense?

Reply:
Per default, if no buckets were found, it returns UNKNOWN. The use can decide if it returns OK or UNKNOWN